File: 백준/Silver/3986. 좋은 단어/좋은 단어.py
N = int(input())

cnt = 0
for n in range(N):
    word = input()
    stack = []
    for i in word:
        if not stack:
            stack.append(i)

        elif stack[-1] == i:
            stack.pop()

        else:
            stack.append(i)

    if not stack:
        cnt += 1

print(cnt)

File: 백준/Silver/3986. 좋은 단어/좋은 단어.cc
#include <iostream>
#include <string>
#include <stack>
using namespace std;

int main() {

	int N;
	cin >> N;
	int answer = 0;
	for (int i = 0;i < N;i++) {
		string str;
		cin >> str;
		stack<char>s;
		for (auto now : str) {
			if (s.empty()) {
				s.push(now);
			}
			else {
				if (s.top() == now) {
					s.pop();
				}
				else{
					s.push(now);
				}
			}
		}
		if (s.empty()) {
			answer++;
		}
	}
	cout << answer;

	return 0;
}